### ISSUE TITLE
Speeding up processing of single audio file with multithreading in faster-whisper

### ISSUE THREAD

**Issue by RomanKlimov** [guillaumekln/faster-whisper#133]
First of all, I want to express my gratitude for your work on the faster-whisper library. It's truly amazing!

I was wondering if it's possible to speed up the processing of a single large audio file using multithreading. I followed the suggestion from this issue: https://github.com/guillaumekln/faster-whisper/issues/100, and tried using the following code:
model = WhisperModel("medium", device="cpu", cpu_threads=8)
However, I didn't observe any speedup while running on my M1 Pro CPU. In fact, starting from cpu_threads=6, the processing time started to increase.

I'm curious about how exactly the speedup per single file using threads works. I've noticed other implementations that achieve speedup by splitting the file into chunks and processing them concurrently, such as https://modal.com/docs/guide/whisper-transcriber and https://github.com/MrEdwards007/WhisperTaskAcceleration.

1. Does splitting the file and processing chunks concurrently make sense for speeding up the transcription process?
2. If yes, is it possible to achieve a similar speedup with faster-whisper out of the box, or would I need to modify the library to implement this functionality?

**Comment by standyyyy** [comment]
Yes, Faster-Whisper is great, I am currently using it, but I also hope to make it faster and process more audio at the same time, check the information and see that ctranslate2 supports batch translation, I don’t know if I have considered adding it What about this function? I recently tried to use TensorRT to run whisper for speed. I'm currently having some difficulties and haven't implemented it yet. Has anyone tried TensorRT to see how fast it is? i really want to know

**Comment by guillaumekln** [comment]
> However, I didn't observe any speedup while running on my M1 Pro CPU. In fact, starting from cpu_threads=6, the processing time started to increase.

On Apple Silicon, the model is running in float32 by default which uses Apple Accelerate. Apple Accelerate does not use multiple threads actually so the option `cpu_threads` should not make a difference.

Starting with `ctranslate2>=3.11` we also support int8 computation on Apple Silicon which does use multiple threads. Try setting `compute_type="int8"`.

> Does splitting the file and processing chunks concurrently make sense for speeding up the transcription process?

Yes, that’s also a good approach. In that case you should increase the parameter `num_workers` for the model to execute multiple task in parallel and run each chunk in a different Python thread.

> If yes, is it possible to achieve a similar speedup with faster-whisper out of the box, or would I need to modify the library to implement this functionality?

Currently there is no option to automatically split the file and parallelize the execution. You would need to develop something on top of faster-whisper.

**Comment by RomanKlimov** [comment]
Thanks for fast reply! I will try to develop and run file chunking on top 

My version of ctranslate2 is exactly 3.11, with compute_type="int8" seems like i'm getting slightly more performance 9.5s vs 10s on the small file. But again, since cpu_threads >=6 time starting to increase and with cpu_threads=10 it's already 22s processing time (x2 default)

Could you please clarify how exactly threads improve the performance in this case and how to properly choose the number of threads?

**Comment by guillaumekln** [comment]
In this case the multiple threads are used to accelerate the matrix multiplications that are executed when running the model. For int8 and Apple Silicon this happens in this library: https://github.com/google/ruy

Depending on the model size, it is possible that the performance gain is small or even negative.

**Comment by Majdoddin** [comment]
There is the small draw-back, that whisper feeds the transcription of the previous 30s window as `prompt` to the next window, to get a continuous text, if you chunk the audio, this feature cannot be optimally used.

**Comment by RomanKlimov** [comment]
@Majdoddin @guillaumekln 
wrote a small wrapper for this purpose, using ffmpeg for detecting silence, in case anyone else need it :) 
https://github.com/RomanKlimov/faster-whisper-acceleration

**Comment by comienzo2093** [comment]
> @Majdoddin @guillaumekln
> wrote a small wrapper for this purpose, using ffmpeg for detecting silence, in case anyone else need it :)
> https://github.com/RomanKlimov/faster-whisper-acceleration

"Thank you, this is very useful. However, wouldn't it be better to split the audio based on the results of VAD?

**Comment by RomanKlimov** [comment]
@comienzo2093 
Thank you for your feedback. 
Maybe it's better to use VAD results, I didn't really try to compare it with ffmpeg results, but I like that ffmpeg gives opportunity to tune volume of silence and duration (not sure about VAD)
I just picked ffmpeg as result of already proven solution described here by this company..
https://modal.com/docs/guide/whisper-transcriber
Let me know if you will try to do it using vad :)

**Comment by wuast94** [comment]
maybe something special in my case, i have a 24c/48t CPU, i see that all cores get used but only by 30%-33%, i dont see that i have any bottleneck, over 100GB ram free, disk isnt used etc.
